libjulia/backends/webassembly/IndentedWriter.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @date 2017
+ * Julia to WebAssembly code generator.
+ */
+
+#include <libjulia/backends/webassembly/IndentedWriter.h>
+#include <libsolidity/interface/Utils.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::solidity;
+
+string IndentedWriter::format() const
+{
+	string result;
+	for (auto const& line: m_lines)
+		result += string(line.indentation * 4, ' ') + line.contents + "\n";
+	return result;
+}
+
+void IndentedWriter::newLine()
+{
+	if (!m_lines.back().contents.empty())
+		m_lines.push_back({ "", m_lines.back().indentation });
```

Comment:
This looks like you only need to store the current indentation and the full text up to (excluding) the current line.